aiida/tools/importexport/dbimport/backends/common.py

```diff
@@ -25,27 +24,47 @@
 MAX_GROUPS = 100
 
 
-def _copy_node_repositories(*, uuids_to_create: List[str], reader: ArchiveReaderAbstract):
+def _copy_node_repositories(*, repository_metadatas: List[str], reader: ArchiveReaderAbstract):
     """Copy repositories of new nodes from the archive to the AiiDa profile.
 
-    :param uuids_to_create: the node UUIDs to copy
+    :param repository_metadatas: list of repository metadatas
     :param reader: the archive reader
-
     """
-    if not uuids_to_create:
+    from disk_objectstore import Container
+    from aiida.manage.manager import get_manager
+
+    if not repository_metadatas:
         return
+
     IMPORT_LOGGER.debug('CREATING NEW NODE REPOSITORIES...')
-    with get_progress_reporter()(total=1, desc='Creating new node repos') as progress:
 
-        _callback = create_callback(progress)
+    reader._extract(path_prefix='')  # pylint: disable=protected-access
+
+    # Copy the contents of the repository container. Note that this should really be done within the
+    # database transaction and if that fails, the written files should be deleted, or at least
+    # soft-deleted. This is to be done later. This is first working example
+
+    container_export = Container(os.path.join(reader._sandbox.abspath, 'container'))  # pylint: disable=protected-access
```

Comment:
as with the writer, control should be inverted here, i.e. the reader should return a Container as part of its API, not the import function accessing protected attributes 😬